core/src/main/java/org/apache/calcite/util/graph/DefaultDirectedGraph.java

```diff
@@ -110,8 +122,8 @@ public E addEdge(V vertex, V targetVertex) {
 
   public E getEdge(V source, V target) {
     // REVIEW: could instead use edges.get(new DefaultEdge(source, target))
-    final VertexInfo<V, E> info = vertexMap.get(source);
-    for (E outEdge : info.outEdges) {
+    final VertexInfo<V, E> info = vertexOutMap.get(source);
+    for (E outEdge : info.edges) {
       if (outEdge.target.equals(target)) {
```

Comment:
Could you please add a test to show the performance improvements?

Reply:
@chunweilei Thanks for your feedback. I will add a benchmark to show the performance difference.

Reply:
@chunweilei I have added a benchmark to measure performance difference. It uses a 8-level binary tree as the graph. The results show that the change significantly improves the performance:

Before:
Benchmark                                              Mode  Cnt   Score   Error  Units
DefaultDirectedGraphBenchmark.getInwardEdgesBenchmark  avgt   25  21.663 ± 0.956  us/op

After:
Benchmark                                              Mode  Cnt  Score    Error  Units
DefaultDirectedGraphBenchmark.getInwardEdgesBenchmark  avgt   25  0.073 ±  0.001  us/op

Reply:
I am not sure I understand these numbers. So the 21.663 and 0.073 are the scores? What do they mean?

Reply:
Also do we have any existing uint test for DefaultDirectedGraph and AttributedDirectedGraph? Do we need to add any new tests?

Reply:
> I am not sure I understand these numbers. So the 21.663 and 0.073 are the scores? What do they mean?

These numbers represent the average time (in microseconds). This could be figured out from the benchmark annotation:

@Benchmark
  @BenchmarkMode(Mode.AverageTime)
  @OutputTimeUnit(TimeUnit.MICROSECONDS)

Reply:
> Also do we have any existing uint test for DefaultDirectedGraph and AttributedDirectedGraph? Do we need to add any new tests?

Currently, we have org.apache.calcite.util.graph.DirectedGraphTest. I am not sure if we need any more tests, as this PR does not change class external behaviors, or introduce new features. If you have something in mind, please let me know. Thanks a lot.